Tests/Packet++Test/Tests/WireGuardTests.cpp

```diff
@@ -0,0 +1,336 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "Packet.h"
+#include "WireGuardLayer.h"
+#include "SystemUtils.h"
+#include <cstring>
+#include "EndianPortable.h"
+
+PTF_TEST_CASE(WireGuardHandshakeInitParsingTest)
+{
+	timeval time;
+	gettimeofday(&time, nullptr);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/WireGuardHandshakeInitiation.dat");
+
+	pcpp::Packet wgHandShakeInitPacket(&rawPacket1);
+
+	PTF_ASSERT_TRUE(wgHandShakeInitPacket.isPacketOfType(pcpp::WireGuard));
+
+	pcpp::WireGuardLayer* wgLayer = wgHandShakeInitPacket.getLayerOfType<pcpp::WireGuardLayer>();
+	PTF_ASSERT_NOT_NULL(wgLayer);
+
+	pcpp::WireGuardHandshakeInitiationLayer* wgHandShakeInitLayer =
+	    wgHandShakeInitPacket.getLayerOfType<pcpp::WireGuardHandshakeInitiationLayer>();
```

Comment:
nit: you can use `auto` here.

Same in the other tests - let's use `auto` wherever possible?